### ISSUE TITLE
Possibility to only specify some keys in the SRC 

### ISSUE THREAD

**Issue by efouladi** [david-janssen/kmonad#41]
I was wondering if it's possible to only specify the keys you want to remap in SRC and sort of pass through the rest of the keys

**Comment by david-janssen** [comment]
Not at the moment, but that's a good idea. It goes on the pile of things to implement. Will get back to you.

**Comment by theol0403** [comment]
I second this, it would be nice to have the rest of my (100%) keyboard work without needing to explicitly map each button just to let them pass through (which would create a really bloated config). For example, even though I have a number layer for my alpha keys, with kmonad enabled (with a map similar to laptop.kbd), the whole number pad does not work at all.

**Comment by david-janssen** [comment]
Alright, I have a test-implementation of this added to refactor, it would be great if you could give it a spin and see if you run into any bugs (also because refactor is my development branch that is *almost* ready, so it would be great if I could get some feedback).

Anyways: on the refactor branch there is now a configuration option 'fallthrough', which defaults to false, which is the standard behavior with which you are already familiar. However, if you set 'fallthrough' to True (see the fallthrough_test.kbd in templates for an example of how), then any keys that fall through all of the layers without ever getting handled simply get emitted.

Next step is creating a new kind of button, at the moment we have a 'block' button, that simply catches an event but does nothing, blocking that key in that layer. We have a 'transparent', which drops the button down into the previous layer, to see if it gets handled. Now we need a third type of transparency/block: If fallthrough is True, then it simply emits the key as-is, if fallthrough is False, it blocks. That way you can turn of handling of keys in layers without overriding your fallthrough behavior. I will post here when this has been added too, and will only close the issue when this is tested and working.

Lemme know how it goes.

**Comment by theol0403** [comment]
Alright, I switched to the refactor branch and tried it out. The fallthrough seems to work great! Thanks a lot for doing this.

However, I noticed a problem that I'm not sure comes from refactor or just this change, which is preventing me from switching to refactor.

In the old docs, it says:
> The tap-hold pauses all event processing until it decides what kind of button it is, and then emits either the tap of its tap-button (when tapping) or the press of its hold button (when holding), with the release coming when the tap-hold is released again.

However, this does not seem to be the case anymore. If I press (and hold) a button with tap-hold, and then type characters (which are supposed to now be from a different layer) within the tap-hold delay, the characters from the base layer get emitted immediately.

Here is my layout for reference: https://github.com/theol0403/kmonad-layout/blob/master/refactor.kbd
